src/Microsoft.Identity.Client/IAuthenticationResult.cs

```diff
@@ -0,0 +1,71 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Contains the results of one token acquisition operation.
+    /// </summary>
+    public interface IAuthenticationResult
+    {
+        /// <summary>
+        /// Gets the Access Token requested.
+        /// </summary>
+        string AccessToken { get; }
+
+        /// <summary>
+        /// Gets the point in time in which the Access Token returned in the Token property ceases to be valid.
+        /// This value is calculated based on current UTC time measured locally and the value expiresIn received from the
+        /// service.
+        /// </summary>
+        DateTimeOffset ExpiresOn { get; }
+
+        /// <summary>
+        /// Gets an identifier for the tenant the token was acquired from. This property will be null if tenant information is
+        /// not returned by the service.
+        /// </summary>
+        string TenantId { get; }
+
+        /// <summary>
+        /// Gets User object. Some elements in User might be null if not returned by the
+        /// service. It can be passed back in some API overloads to identify which user should be used.
+        /// </summary>
+        User User { get; }
+
+        /// <summary>
+        /// Gets the entire Id Token if returned by the service or null if no Id Token is returned.
+        /// </summary>
+        string IdToken { get; }
+
+        /// <summary>
+        /// Gets the scope values returned from the service.
+        /// </summary>
+        string[] Scope { get; }
```

Comment:
Or just IEnumerable<>, depending on what's available in all of the target platforms. Either way, it's a separate change.